### ISSUE TITLE
Make possible to lock the ability to expand the menu only by clicking on the expand icon

### ISSUE THREAD

**Issue by eduardobalbo** [ant-design/ant-design#30554]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
The default behaviour of the Menu component and submenu now is, when you click the text. or anywhere on the menu itself, it expands the childrens of that menu.
In our use case, we have a target on the menu that opens a modal, and a target on the submenu that opens a new tab, so clicking on the menu title opens the modal or the tab but also expands the menu and reveals the childrens, we want to expand the menu, only when the expanding icon is clicked.
This would give the ability to control wether the user wants to expand the menu, or trigger the action of the menu itself by clicking the title or the component.
In short, locking the ability to expand the menu and its children to the expand icon.

### What does the proposed API look like?
```
Type expandOn = {
 'icon-click' | 'text-click' | 'default' 
}
```

```
<Menu 
 expandOn={'icon-click'}
>
  <Menu.Item>Menu</Menu.Item>
  <SubMenu title="SubMenu">
    <Menu.Item>SubMenuItem</Menu.Item>
  </SubMenu>
</Menu>
```

**Comment by afc163** [comment]
You may have to create your own menu accourding your UX design.

**Comment by eduardobalbo** [comment]
Is there any reason why this is not even tought about? Is it way to hard to implement/time demanding or the actual structure of the components don't allow this kind of change?
I would appreciate any ideas on how to achieve this with Antd. We dont want to mix packages on our project.
